File: Leetcode/112. Path Sum.py
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root is None:
            return False
        targetSum -= root.val
        
        if targetSum == 0:
            if (root.left is None) and (root.right is None):
                return True
        return self.hasPathSum(root.left, targetSum) or self.hasPathSum(root.right, targetSum)

File: Leetcode/112. Path Sum.java
class Solution {
    public boolean fun(TreeNode root, int targetSum, int sum){
        if(root == null)
            return false;
        if(root.left == null && root.right == null){
            sum = sum + root.val;
            if(sum == targetSum)
                return true;   
        }
        return fun(root.left, targetSum, sum + root.val) || fun(root.right, targetSum, sum + root.val);
    }
    public boolean hasPathSum(TreeNode root, int targetSum) {
        int sum = 0;
        return fun(root, targetSum, sum);
    }
}
